Update tallyarbiter-m5stickc.ino

Fred-DTV opened the pull request:
1.  added to option for user defined colours (Grey is used for the font on the idle screen to save battery)
2. Added full set of network settings. especially for environments where static IPs are necessary
3. added was_pressed for easier used of the buttons
4. CPU frequency lowered and btstop added to save battery
5. Added a automatic switch to the tally screen at boot to have the Tally straight into a working state for an event in case it needs to be restartet
6. Added "charging..." log to see if it is plugged into power

Fred-DTV committed:
Update tallyarbiter-m5stickc.ino

1.  added to option for user defined colours (Grey is used for the font on the idle screen to save battery)
2. Added full set of network settings. especially for environments where static IPs are necessary
3. added was_pressed for easier used of the buttons
4. CPU frequency lowered and btstop added to save battery
5. Added a automatic switch to the tally screen at boot to have the Tally straight into a working state for an event in case it needs to be restartet
6. Added "charging..." log to see if it is plugged into power
7.

Fred-DTV commented:
This is my very first pull request. I hope I did it right.

josephdadams commented:
This is great!

A couple of suggestions:

Add a boolean variable `USE_STATIC` up in the configuration area. If `true`, then it applies the static IP information as you've programmed. If `false`, then it doesn't, and uses DHCP like previously. Otherwise, users will have to comment these lines out. Default to `false`.

I also recommend we remove this line:
https://github.com/DachsbauTV/TallyArbiter-M5StickCListener/blob/16deecd153e02800b179cad1e4ef6e7653c6a237/tallyarbiter-m5stickc.ino#L97

And instead call that function instead inside `socket_DeviceId` around line 230. This way, once the M5 is successfully connected to the server, the server will emit this event, signifying a valid connection. As you have it implemented currently, it delays for 10 seconds which is _probably_ enough time, but might be too much sometimes.

Fred-DTV committed:
Update tallyarbiter-m5stickc.ino

1. It now just switches straight to tally mode when wifi and server connection works
2. I can't get DHCP back to work. Even if USE_STATIC is false it will still connect with the set clientIP. Any ideas on this?
3. This doesn't return the actual IP address. It only shows a value in the log screen: 
void WiFiEvent(WiFiEvent_t event) {
  switch (event) {
    case SYSTEM_EVENT_STA_GOT_IP:
      logger("Network connected!", "info");
      logger(String(WiFi.localIP()), "info");
      networkConnected = true;
      break;
